### ISSUE TITLE
Error parsing SQL: 'DELETE FROM

### ISSUE THREAD

**Issue by eugenbg** [zendesk/maxwell#439]
Getting this error on launching maxwell
bin/maxwell --user='maxwell' --password='XXXXXX' --host='127.0.0.1' --producer=stdout

11:28:38,560 ERROR SchemaChange - Error parsing SQL: 'DELETE FROM `magento2dockerized`.`catalog_category_product_index_tmp`'
com.zendesk.maxwell.schema.ddl.MaxwellSQLSyntaxError: DELETE
        at com.zendesk.maxwell.schema.ddl.MysqlParserListener.visitErrorNode(MysqlParserListener.java:84)
        at org.antlr.v4.runtime.tree.ParseTreeWalker.walk(ParseTreeWalker.java:41)
        at org.antlr.v4.runtime.tree.ParseTreeWalker.walk(ParseTreeWalker.java:52)
        at com.zendesk.maxwell.schema.ddl.SchemaChange.parseSQL(SchemaChange.java:79)
        at com.zendesk.maxwell.schema.ddl.SchemaChange.parse(SchemaChange.java:91)
        at com.zendesk.maxwell.schema.AbstractSchemaStore.resolveSQL(AbstractSchemaStore.java:48)
        at com.zendesk.maxwell.schema.MysqlSchemaStore.processSQL(MysqlSchemaStore.java:82)
        at com.zendesk.maxwell.MaxwellReplicator.processQueryEvent(MaxwellReplicator.java:415)
        at com.zendesk.maxwell.MaxwellReplicator.getRow(MaxwellReplicator.java:384)
        at com.zendesk.maxwell.MaxwellReplicator.work(MaxwellReplicator.java:146)
        at com.zendesk.maxwell.RunLoopProcess.runLoop(RunLoopProcess.java:35)
        at com.zendesk.maxwell.Maxwell.start(Maxwell.java:163)
        at com.zendesk.maxwell.Maxwell.main(Maxwell.java:180)

**Comment by osheroff** [comment]
this generally means that you have a client connected from before you moved to row-based replication -- for whatever reason this config flag applies to the connection, not the server level. 

see http://maxwells-daemon.io/quickstart/#row-based-replication

**Comment by liteonly** [comment]
@osheroff 
Hey, i am facing this issue while trying to use maxwell.
```
11:53:51,820 DEBUG SchemaChange - SQL_PARSE <- "DELETE FROM `dbsync`.`universal_location_search_mem`"
11:53:51,893 ERROR MysqlParserListener - (parse DELETE FROM `dbsync` . `universal_location_search_mem`)
11:53:51,894 ERROR SchemaChange - Error parsing SQL: 'DELETE FROM `dbsync`.`universal_location_search_mem`'
11:53:51,895 DEBUG TransportImpl - disconnected from 127.0.0.1:3306
com.zendesk.maxwell.schema.ddl.MaxwellSQLSyntaxError: DELETE
	at com.zendesk.maxwell.schema.ddl.MysqlParserListener.visitErrorNode(MysqlParserListener.java:84)
	at org.antlr.v4.runtime.tree.ParseTreeWalker.walk(ParseTreeWalker.java:41)
	at org.antlr.v4.runtime.tree.ParseTreeWalker.walk(ParseTreeWalker.java:52)
	at com.zendesk.maxwell.schema.ddl.SchemaChange.parseSQL(SchemaChange.java:79)
	at com.zendesk.maxwell.schema.ddl.SchemaChange.parse(SchemaChange.java:91)
	at com.zendesk.maxwell.schema.AbstractSchemaStore.resolveSQL(AbstractSchemaStore.java:47)
	at com.zendesk.maxwell.schema.MysqlSchemaStore.processSQL(MysqlSchemaStore.java:82)
	at com.zendesk.maxwell.replication.MaxwellReplicator.processQueryEvent(MaxwellReplicator.java:473)
	at com.zendesk.maxwell.replication.MaxwellReplicator.getRow(MaxwellReplicator.java:443)
	at com.zendesk.maxwell.replication.MaxwellReplicator.work(MaxwellReplicator.java:159)
	at com.zendesk.maxwell.util.RunLoopProcess.runLoop(RunLoopProcess.java:35)
	at com.zendesk.maxwell.Maxwell.start(Maxwell.java:167)
	at com.zendesk.maxwell.Maxwell.main(Maxwell.java:188)
11:53:51,970 INFO  Maxwell - starting shutdown
11:53:51,971 DEBUG MysqlPositionStore - writing heartbeat: 1482215029612
```

This is the structure for the table universal_location_search_mem
```
CREATE TABLE `universal_location_search_mem` (
  `location_id` int(11) NOT NULL,
  `location_string` varchar(255) NOT NULL
) ENGINE=MEMORY DEFAULT CHARSET=latin1;
```

I think, this error is happening for tables with engine as 'memory', where the delete event from these temp tables is written to the master mysql bin log file, not in the binlog format, but the query itself
This is the binlog entry in the master bin file
```
SET TIMESTAMP=1482215030/*!*/;
DELETE FROM `dbsync`.`universal_location_search_mem`
/*!*/;
# at 1197
#161220 11:53:51 server id 1  end_log_pos 1272 CRC32 0xb8cda7c8         Query   thread_id=5     exec_time=0     error_code=0
SET TIMESTAMP=1482215031/*!*/;
```

**Comment by osheroff** [comment]
interesting. 

> When a master server shuts down and restarts, its MEMORY tables become empty. To replicate this effect to slaves, the first time that the master uses a given MEMORY table after startup, it logs an event that notifies slaves that the table must to be emptied by writing a DELETE statement for that table to the binary log.

**Comment by osheroff** [comment]
fixed in 1.6.0
